neurots/utils.py

```diff
@@ -41,8 +44,37 @@ def format_values(obj, decimals=None):
     return obj
 
 
-def _check(data):
-    """Checks if data in dictionary are empty."""
-    for key, val in data.items():
-        if len(val) == 0:
-            raise NeuroTSError(f"Empty distribution for diameter key: {key}")
+def _warn(key):
+    warnings.warn(
+        f"The '{key}' property is deprecated, please use '{key}_dendrite' instead",
+        DeprecationWarning,
+    )
+
+
+def convert_from_legacy_neurite_type(data):
+    """Convert legacy neurite type names, basal -> basal_dendrite and apical -> apical_dendrite."""
+    old_data = deepcopy(data)
+    for key, _data in old_data.items():
+
+        if key == "apical":
+            _warn(key)
+            data["apical_dendrite"] = data.pop("apical")
+            key = "apical_dendrite"
+        if key == "basal":
+            _warn(key)
+            data["basal_dendrite"] = data.pop("basal")
+            key = "basal_dendrite"
+
+        if isinstance(_data, dict):
+            data[key] = convert_from_legacy_neurite_type(data[key])
+
+        if isinstance(_data, list):
+            for i, d in enumerate(_data):
+                if d == "apical":
+                    _warn(d)
+                    data[key][i] = "apical_dendrite"
+                if d == "basal":
+                    _warn(d)
+                    data[key][i] = "basal_dendrite"
```

Comment:
Why not using recursion here as for dict values?

Reply:
You mean to deal with a possible list of lists? There are no such things in these files, no?

Reply:
Yeah I don't think so indeed.